src/main/java/org/opentripplanner/gtfs/mapping/GTFSToOtpTransitServiceMapper.java

```diff
@@ -133,28 +152,60 @@ private void mapAndAddTransfersToBuilder() {
         builder.getTransfers().addAll(transferMapper.map(data.getAllTransfers()));
     }
 
+    private Set<String> getLanguagesFromTranslations(Collection allTranslations) {
+        Set<String> languages = new HashSet<>(Set.of());
+        java.util.Iterator<org.onebusaway.gtfs.model.Translation> iterator =
+                allTranslations.iterator();
+        while (iterator.hasNext()) {
+            org.onebusaway.gtfs.model.Translation translation = iterator.next();
+            languages.add(translation.getLanguage());
+        }
+        return languages;
+    }
+
+    private I18NString getTranslations(
+            TranslationServiceImpl ts,
+            Set<String> languages,
+            Object object,
+            String fieldName
+    ) {
+        HashMap<String, String> translations = new HashMap<>();
+        for (String lang: languages) {
+            if (object instanceof org.onebusaway.gtfs.model.Stop) {
+                org.onebusaway.gtfs.model.Stop t = ts.getTranslatedEntity(lang, org.onebusaway.gtfs.model.Stop.class,
```

Comment:
This is an awfully ineffective way to get the translations, especially if you look at how it is implemented, would it make sense to directly read in and index all the translations ourselves?

Reply:
Also this might be better be encapsulated in a separate class, which is passed in to the mappers

Reply:
Created new TranslationHelper.